pkg/syncthing/api.go

```diff
@@ -26,14 +27,14 @@ func NewAPIClient() *http.Client {
 }
 
 // APICall calls the syncthing API and returns the parsed json or an error
-func (s *Syncthing) APICall(url, method string, code int, params map[string]string, local bool) ([]byte, error) {
+func (s *Syncthing) APICall(url, method string, code int, params map[string]string, local bool, body []byte) ([]byte, error) {
 	var urlPath string
 	if local {
 		urlPath = path.Join(s.GUIAddress, url)
 	} else {
 		urlPath = path.Join(s.RemoteGUIAddress, url)
 	}
-	req, err := http.NewRequest(method, fmt.Sprintf("http://%s", urlPath), nil)
+	req, err := http.NewRequest(method, fmt.Sprintf("http://%s", urlPath), bytes.NewBuffer(body))
```

Comment:
should we check if we get a GET/DELETE with a non nil body?

Reply:
no, nil argument is fine